extension/background.js

```diff
@@ -1,18 +1,54 @@
 var port = chrome.runtime.connectNative("default_opener");
 
+/**
+ * Call the native appliction with this url to open it with the default browser
+ */
+function callUrl(url) {
+    console.log(`Sending: link ${url} `);
+    port.postMessage({link: url});
+}
+
 port.onMessage.addListener((response) => {
-  console.log("Received: " + response);
+    console.log("Received: " + response);
 });
 
 port.onDisconnect.addListener(() => {
-  console.log("Port was disconnected. Error:", chrome.runtime.lastError);
+    console.log("Port was disconnected. Error:", chrome.runtime.lastError);
+    /* If we lost connection, simply reconnect. */
+    port = chrome.runtime.connectNative("default_opener");
 });
 
+/** Create a new context menu to open links with an external browser */
 chrome.contextMenus.create({
   title: 'Open in default browser',
   contexts: ['link'],
   onclick: (info) => {
-    console.log(`Sending: link ${info.linkUrl} `);
-    port.postMessage({link: info.linkUrl});
+      callUrl(info.linkUrl);
   }
 });
+
+/** I have no idea what this does */
```

Comment:
[browserAction.onClick](https://developer.chrome.com/extensions/browserAction#event-onClicked) gets triggered when you click the extension icon in the Chrome toolbar. I can see how this would be useful.